### PR TITLE
Add IT for installation from cache

### LINKED ISSUE #77: Composer - remote repository
Please support remote repository in composer-adapter for proxying packagist.org. 

Progress:
- [x] basic implementation 
- [x] unit tests for added classes
- [x] IT for installation from remote
- [x] IT for installation from local

Caching of combined from local and remote files
- [x] class for validate cached item by checking time of saving on expiration `CacheTimeControl`
- [x] implementation of `Cache` interface for this repository
- [x] tests for `ComposerStorageCache`
- [ ] IT for installation from cache

### PR DESCRIPTION
**Pull request by genryxy**
Closes #77 
Added IT case for installation from cache. Made class `ComposerStorageCache` public for using in `artipie` module

### CODE REVIEW

File: src/test/java/com/artipie/composer/http/proxy/CacheComposerIT.java [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+        if (this.sourceserver != null) {
+            this.sourceserver.close();
+        }
+        this.client.stop();
+        this.cntn.stop();
+        try {
+            FileUtils.cleanDirectory(this.tmp.toFile());
+            Files.deleteIfExists(this.tmp);
+        } catch (final IOException ex) {
+            ex.printStackTrace();
```
Comment (olenagerasimova) [inline src/test/java/com/artipie/composer/http/proxy/CacheComposerIT.java:161]: @genryxy hm, I'd suggest to use `Logger.error()` here to print exception

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@genryxy thanks, just one suggestion

**Comment by genryxy** [comment]
@olenagerasimova thanks, corrected

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@genryxy thanks, one more suggestions: let's, please, remove unnecessary suppression
`@SuppressWarnings("PMD.AvoidPrintStackTrace")`

**Comment by genryxy** [comment]
@olenagerasimova removed unnecessary suppression

**Review by olenagerasimova — APPROVED** [review]
@genryxy thanks

**Pull request merged**